TiDB panic when start with security.skip-grant-table

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->

1. Start a TiDB instance with configuration `security.skip-grant-table` and OS root privilege.
2. Connect to TiDB with `mysql -h127.1 -uroot -P4000`

### 2. What did you expect to see? (Required)

Connect successfully

### 3. What did you see instead (Required)

TiDB panic

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x1047d5768]

goroutine 4862 [running]:
github.com/pingcap/tidb/pkg/privilege/privileges.(*Handle).ensureActiveUser(0x0, {0x1074bc080, 0x10aefc848}, {0x14008e3e084, 0x4})
	/Users/zhangming/gopath/src/github.com/pingcap/tidb/pkg/privilege/privileges/cache.go:2122 +0x98
github.com/pingcap/tidb/pkg/privilege/privileges.(*UserPrivileges).GetUserResources(0x14008df6fc0, {0x14008e3e084, 0x4}, {0x14008e3e020, 0x9})
	/Users/zhangming/gopath/src/github.com/pingcap/tidb/pkg/privilege/privileges/privileges.go:326 +0x6c
github.com/pingcap/tidb/pkg/server.(*clientConn).checkUserConnectionCount(0x14008e0a820, {0x14008e3e020, 0x9})
	/Users/zhangming/gopath/src/github.com/pingcap/tidb/pkg/server/user_connections.go:118 +0x1b8
github.com/pingcap/tidb/pkg/server.(*clientConn).openSessionAndDoAuth(0x14008e0a820, {0x14008c2b3a6, 0x0, 0x90}, {0x14008935380, 0x15}, 0x0)
	/Users/zhangming/gopath/src/github.com/pingcap/tidb/pkg/server/conn.go:808 +0x18c
github.com/pingcap/tidb/pkg/server.(*clientConn).readOptionalSSLRequestAndHandshakeResponse(0x14008e0a820, {0x1074bbfa0, 0x14008e40180})
	/Users/zhangming/gopath/src/github.com/pingcap/tidb/pkg/server/conn.go:644 +0x1288
github.com/pingcap/tidb/pkg/server.(*clientConn).handshake(0x14008e0a820, {0x1074bbfa0, 0x14008e40180})
	/Users/zhangming/gopath/src/github.com/pingcap/tidb/pkg/server/conn.go:320 +0x21c
github.com/pingcap/tidb/pkg/server.(*Server).onConn(0x14003e4a120, 0x14008e0a820)
	/Users/zhangming/gopath/src/github.com/pingcap/tidb/pkg/server/server.go:671 +0x6e4
created by github.com/pingcap/tidb/pkg/server.(*Server).startNetworkListener in goroutine 675
	/Users/zhangming/gopath/src/github.com/pingcap/tidb/pkg/server/server.go:551 +0x980
```

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

master

--- a/pkg/privilege/privileges/privileges.go
+++ b/pkg/privilege/privileges/privileges.go
@@ -323,6 +323,9 @@ func (p *UserPrivileges) isValidHash(record *UserRecord) bool {
 
 // GetUserResources gets the maximum number of connections for the current user
 func (p *UserPrivileges) GetUserResources(user, host string) (int64, error) {
+	if SkipWithGrant {
+		return 0, nil
+	}
 	terror.Log(p.Handle.ensureActiveUser(context.Background(), user))
 	mysqlPriv := p.Handle.Get()
 	record := mysqlPriv.connectionVerification(user, host)

--- a/pkg/privilege/privileges/privileges_test.go
+++ b/pkg/privilege/privileges/privileges_test.go
@@ -2083,7 +2083,7 @@ func TestVerificationInfoWithSessionTokenPlugin(t *testing.T) {
 	require.ErrorContains(t, err, "Access denied")
 }
 
-func TestNilHandleInConnectionVerification(t *testing.T) {
+func TestNilHandleInSkipWithGrant(t *testing.T) {
 	config.GetGlobalConfig().Security.SkipGrantTable = true
 	privileges.SkipWithGrant = true
 	defer func() {
@@ -2092,7 +2092,13 @@ func TestNilHandleInConnectionVerification(t *testing.T) {
 	}()
 	store := testkit.CreateMockStore(t)
 	tk := testkit.NewTestKit(t, store)
+	// check ConnectionVerification
 	require.NoError(t, tk.Session().Auth(&auth.UserIdentity{Username: "root", Hostname: `%`}, nil, nil, nil))
+	// check GetUserResources
+	pc := privilege.GetPrivilegeManager(tk.Session())
+	userLimit, err := pc.GetUserResources("root", "%")
+	require.NoError(t, err)
+	require.EqualValues(t, 0, userLimit)
 }
 
 func testShowGrantsSQLMode(t *testing.T, tk *testkit.TestKit, expected []string) {

